src/main/java/mate/academy/internetshop/dao/jdbc/ShoppingCartDaoJdbcImpl.java

```diff
@@ -0,0 +1,116 @@
+package mate.academy.internetshop.dao.jdbc;
+
+import java.sql.Connection;
+import java.sql.PreparedStatement;
+import java.sql.ResultSet;
+import java.sql.SQLException;
+import java.sql.Statement;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Optional;
+import mate.academy.internetshop.dao.Dao;
+import mate.academy.internetshop.dao.ProductDao;
+import mate.academy.internetshop.dao.ShoppingCartDao;
+import mate.academy.internetshop.lib.Inject;
+import mate.academy.internetshop.model.Product;
+import mate.academy.internetshop.model.ShoppingCart;
+import mate.academy.internetshop.util.ConnectionUtil;
+
+@Dao
+public class ShoppingCartDaoJdbcImpl implements ShoppingCartDao {
+    @Inject
+    private ProductDao productDao;
+
+    @Override
+    public ShoppingCart create(ShoppingCart element) {
+        try (Connection con = ConnectionUtil.getConnection()) {
+            String sql = "INSERT INTO shopping_carts\n"
+                    + "(user_id) VALUES (?);";
+            PreparedStatement ps = con.prepareStatement(sql, Statement.RETURN_GENERATED_KEYS);
+            ps.setLong(1, element.getUserId());
+            ps.executeUpdate();
+            ResultSet rs = ps.getGeneratedKeys();
+            rs.next();
+            element.setId(rs.getLong(1));
+        } catch (SQLException e) {
+            throw new RuntimeException(e);
+        }
+        return element;
+    }
+
+    @Override
+    public Optional<ShoppingCart> get(Long id) {
+        try (Connection con = ConnectionUtil.getConnection()) {
+            String sql = "SELECT * FROM internet_shop.shopping_carts\n"
```

Comment:
```suggestion
            String sql = "SELECT * FROM shopping_carts\n"
```
по всьому проекту глянь

Reply:
removed everywhere.